src/components/single-line-diagram.js

```diff
@@ -71,17 +78,19 @@ const SingleLineDiagram = forwardRef((props, ref)  => {
     const svgPrevViewbox = useRef();
     const svgDraw = useRef();
 
-    const [forceState, updateState] = React.useState(false);
+    const [forceState, updateState] = useState(false);
+    const [updateSwitchMsg, setUpdateSwitchMsg] = useState("");
+
     const forceUpdate = React.useCallback(() => {
             if (svgDraw.current) {
                 svgPrevViewbox.current = svgDraw.current.viewbox();
             }
             updateState(s=>!s)
     }, []);
 
-
     useImperativeHandle(ref, () => ({
-        reloadSvg: forceUpdate
+        reloadSvg: forceUpdate,
+        updateSwitchMsg: (m) => setUpdateSwitchMsg(m)
```

Comment:
unecessary use of lambda ?

Reply:
Wouldn't it be better to pass the message in a prop ? We are using an imperativeHandle for the refresh because we want it to redownload the exact same url and props don't allow that. But for the message, we do things only when the message changes, so props are more straightforward

Reply:
You're right : using a prop here makes sense. The modification is done